src/benchmarks/micro/libraries/Microsoft.Extensions.DependencyInjection/TimeToFirstServiceBenchmark.cs

```diff
@@ -119,5 +104,90 @@ private class B
         }
 
         private class C { }
+
+        private ServiceProvider BuildServiceProvider(IServiceCollection services)
+        {
+            if (_mode == ServiceProviderMode.Default)
+            {
+                return services.BuildServiceProvider();
+            }
+
+            Assembly asm = typeof(ServiceProvider).Assembly;
+
+            Type serviceProviderEngineType = asm.GetType("Microsoft.Extensions.DependencyInjection.ServiceLookup.ServiceProviderEngine", throwOnError: true, ignoreCase: false);
+            if (serviceProviderEngineType == null)
+            {
+                throw new Exception($"Unable to find ServiceProviderEngine type.");
+            }
```

Comment:
This Type doesn't appear to be used. Is this code necessary?

Reply:
Removed. Thanks